### ISSUE TITLE
[Bug] Calling Launcher. TryOpenAsync(string uri) causes exception

### ISSUE THREAD

**Issue by mbalous** [xamarin/Essentials#1158]
### Steps to Reproduce
1. Invoke this code 
```csharp
string email = "hello@world.com";
await Device.InvokeOnMainThreadAsync(() => Launcher.TryOpenAsync($"mailto:{email}"));
```

### Expected Behavior
No exception is beign thrown

### Actual Behavior
Exception `System.InvalidOperationException` with message _'A method was called at an unexpected time.'_ is thrown 

### Basic Information

- Version with issue: 1.3.1 & 1.5.1
- IDE: VS 2019
- Platform Target Frameworks: 
  - UWP:  Windows 10, version 1809 (10.0; Build 17763)

### Screenshots
![obrazek](https://user-images.githubusercontent.com/10358985/76198752-eaf11400-61ee-11ea-9a2d-8d577b09f3d8.png)

**Comment by jamesmontemagno** [comment]
I am unable to reproduce this with your code. can you please provide a sample.

**Comment by mbalous** [comment]
Go to `TrackerViewModel.MailTo(object obj)` to see the command executed (or place breakpoint). Execute command by launching the application and pressing the button _"Open email uri"_

Executed code on button press:
```csharp
private async void MailTo(object obj)
{
	string email = "hello@world.com";
	await Device.InvokeOnMainThreadAsync(() => Xamarin.Essentials.Launcher.TryOpenAsync($"mailto:{email}"));
}
```

[Sample.zip](https://github.com/xamarin/Essentials/files/4324035/LocationTracker.zip)

.csproj
```xml
<TargetPlatformVersion>10.0.18362.0</TargetPlatformVersion>
<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
.
.
.
<PackageReference Include="Xamarin.Forms" Version="4.5.0.356" />
<PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform" Version="6.2.9" />
<PackageReference Include="Xamarin.Essentials" Version="1.5.1" />
```

**Comment by jamesmontemagno** [comment]
I was able to reproduce your issue. 

Calling "OpenAsync" works fine, however "TryOpenAsync" does not.

Workaround:'

```
private async void MailTo(object obj)
		{
			string email = "hello@world.com";
			if (await Xamarin.Essentials.Launcher.CanOpenAsync("mailto:"))
				await Xamarin.Essentials.Launcher.OpenAsync($"mailto:{email}");
		}
```

You do not need to invoke on the main thread as it is coming from a command in which it already is on the main thread :)

**Comment by jamesmontemagno** [comment]
Hmmm strange. it only seems to be an issue when targeting newer versions of UWP which is why we didn't catch it.

**Comment by mbalous** [comment]
> You do not need to invoke on the main thread as it is coming from a command in which it already is on the main thread :)
Thank you, I'm aware of that, but in order to try fixing this error, I tried various approaches.
Nevertheless, thanks for suggesting the workaround.

**Comment by maexsp** [comment]
Workaround with subject and body (unfortionatly only plaintext on android possible). This opens less non-email related apps to send an email (if using e.g. Email.ComposeAsync(...)  )
`            if (await Launcher.CanOpenAsync(new Uri("mailto:")))
            {
                await Launcher.OpenAsync($"mailto:{emailTo}?subject={subject}&body={body}");
            }
            else
            {
                await Email.ComposeAsync(subject, body, new []{ emailTo });
            }
`

**Comment by Redth** [comment]
Fixed in #1171
